staging/src/k8s.io/apiserver/pkg/endpoints/installer.go

```diff
@@ -839,7 +839,10 @@ func (n rootScopeNaming) GenerateLink(req *restful.Request, obj runtime.Object)
 			return "", err
 		}
 	}
-	return n.pathPrefix + url.QueryEscape(name) + n.pathSuffix, nil
```

Comment:
I'd rather see the suffix built to match.  This code changes behavior for all subresources and will bork things like `/status` and `/scale`.

@lavalamp you were wondering about why we should split out TPRs.  It's because of stuff like this.  There are hacks all over for it and it creates an inscrutable mess.